traitlets/config/loader.py

```diff
@@ -715,10 +792,14 @@ def load_config(self, argv=None, aliases=None, flags=None):
                     lhs = aliases[lhs]
                 if '.' not in lhs:
                     # probably a mistyped alias, but not technically illegal
-                    self.log.warning("Unrecognized alias: '%s', it will probably have no effect.", raw)
+                    self.log.warning("Unrecognized alias: '%s', it will have no effect.", raw)
+                    trait = None
+                else:
+                    trait = self._find_trait(lhs)
                 try:
-                    self._exec_config_str(lhs, rhs)
+                    self._exec_config_str(lhs, rhs, trait)
                 except Exception:
+                    raise
```

Comment:
Good reminder. This is a debug bit because I think this shouldn't raise anymore unless there's a bug. It should be gone when this is done.